src/app/designationEditor/designationEditor.component.js

```diff
@@ -156,12 +156,16 @@ class DesignationEditorController {
       resolve: {
         giveDomain: () => { return this.giveDomain },
         designationNumber: () => { return this.designationNumber },
-        givingLinks: () => { return this.designationContent.givingLinks }
+        designationType: () => { return this.designationContent.designationType },
+        hasNewsletter: this.designationEditorService.hasNewsletter(this.designationNumber),
```

Comment:
Is there a special reason for this to not be a function like the others?

Reply:
yes, it returns a promise, and the others are scalar values. resolve will automatically await the promise.